src/lib/DefaultRuleManager.ts

```diff
@@ -40,8 +40,22 @@ export class DefaultRuleManager implements RuleManager {
 		this.initialized = true;
 	}
 
-	getRulesMatchingCriteria(filters: RuleFilter[]): Rule[] {
-		return this.catalog.getRulesMatchingFilters(filters);
+	/**
+	 * Returns rules matching the filter criteria provided, and any non-conflicting implicit filters.
+	 * @param {RuleFilter[]} filters - A collection of filters.
+	 */
+	async getRulesMatchingCriteria(filters: RuleFilter[]): Promise<Rule[]> {
+		const engineNames: string[] = (await this.resolveEngineFilters(filters)).map(e => e.getName());
+		const rules: Rule[] = this.catalog.getRulesMatchingFilters(filters);
+		return rules.filter(r => engineNames.includes(r.engine));
+	}
+
+	/**
+	 * Returns rules that match only the provided filters, completely ignoring any implicit filtering.
+	 * @param filters
+	 */
+	getRulesMatchingOnlyExplicitCriteria(filters: RuleFilter[]): Promise<Rule[]> {
```

Comment:
Yeah, it could. I chose to make it a Promise because it seems for the `getRulesMatchingCriteria` and `getRulesMatchingOnlyExplicitCriteria` methods to have the same signature. I can change it if you prefer.